File: estudos-pf/nivel-1-map-filter-reduce/solution/q4/q4.clj
(defn len [palavra] 
  (count palavra))

(defn maior? [maior x]
  (if (> maior x) maior x))

(defn pega-maior [sequencia]
  (reduce maior? sequencia))

(defn maior-tamanho [sequencia]
  (pega-maior (map len sequencia)))

(assert (= (maior-tamanho ["a" "bb" "ccc"]) 3))
(assert (= (maior-tamanho ["oi"]) 2))
(assert (= (maior-tamanho ["um" "dois" "tres" "quatro"]) 6))

File: estudos-pf/nivel-1-map-filter-reduce/solution/q4/q4.py
def maior_tamanho(palavras: list) -> int:
    return max(map(len, palavras))

assert maior_tamanho(["a", "bb", "ccc"]) == 3
assert maior_tamanho(["oi"]) == 2
assert maior_tamanho(["um", "dois", "tres", "quatro"]) == 6
